eth_bridge/ic/src/eth_proxy/src/api/burn.rs

```diff
@@ -67,17 +68,20 @@ async fn burn(eth_addr: EthereumAddr, amount: Nat) -> TxReceipt {
                                     weth_ic_addr_pid,
                                     current_balance - amount.clone(),
                                 );
+                                s.remove_user_flag(caller);
+                            });
 
-                                s.add_claimable_message(ClaimableMessage {
+                            ic_kit::ic::spawn(async move {
```

Comment:
there is a `insert_sync` which guarantees finality by doing the spawn for you

https://github.com/Psychedelic/cap/blob/main/sdk/rust/src/transaction/insert.rs#L126